plyer/platforms/android/smsreceive.py

```diff
@@ -0,0 +1,55 @@
+'''
+Android Recieve SMS
+-----------
+'''
+
+from jnius import autoclass
+from plyer.facades import SmsReceive
+from jnius import PythonJavaClass
+from jnius import java_method
+
+
+Intent = autoclass('android.content.Intent')
+Bundle = autoclass('android.os.Bundle')
+GenericBroadcastReceiver = autoclass(
+    'org.renpy.android.GenericBroadcastReceiver'
+)
+SmsMessage = autoclass('android.telephony.SmsMessage')
+Toast = autoclass('android.widget.Toast')
+
+
+class BroadcastReceiver(PythonJavaClass):
+    __javainterfaces__ = [
+        'org/renpy/android/GenericBroadcastReceiverCallback']
+    __javacontext__ = 'app'
+
+    def __init__(self, *args, **kwargs):
+        PythonJavaClass.__init__(self, *args, **kwargs)
+
+    @java_method(
+        '(Landroid/content/Context;Landroid/content/Intent;)V')
+    def onReceive(self, context, intent):
+        bundle = intent.getExtras()
+        messages = None
+        string = ""
+        if bundle:
+            pdus = bundle.get('pdus')
+            messages = SmsMessage[len(pdus)]
+            for i in range(0, len(messages)):
```

Comment:
`range(len(...))` seem to be discouraged by Python community. Personally, I don't like only the zero, as it always starts from zero unless stated otherwise.